src/resty/gcp/init.lua

```diff
@@ -51,50 +51,116 @@ FindApis = function(apiClass, methods, curr)
     return methods
 end
 
+local function template_expansion(str, params)
+    return (string.gsub(str, "{(.-)}", function (key)
+        return ngx.escape_uri(params[key])
+    end))
+end
+
+local function build_request(accesstoken, apiDetail, baseUrl, params, requestBody, opt)
```

Comment:
I agree about the additional `opt` table seeming a little odd in this case. Seems like the caller should just set `params.alt = "media"` before invoking the API.